src/renderer/prefComponents/imageUploader/index.vue

```diff
@@ -99,8 +116,25 @@ export default {
         return this.$store.state.preferences.githubToken
       }
     },
+    prefCliScript: {
+      get: function () {
+        return this.$store.state.preferences.cliScript
+      }
+    },
     githubDisable () {
       return !this.githubToken || !this.github.owner || !this.github.repo
+    },
+    cliScriptDisable () {
+      if (!this.cliScript) {
+        return true
+      }
+      try {
+        const stat = statSync(this.cliScript)
+        return !stat.isFile() || (stat.mode & (S_IXUSR | S_IXGRP | S_IXOTH)) === 0
```

Comment:
Could you please move the code inside try/catch to `utils/filesystem.js` to separate fronted and "backend" stuff. Like `isFileExecutable` and negate the result.